Front-End/challenge2/script.js

```diff
@@ -1,7 +1,93 @@
-/*
-TO-DO:
+const calculatorApp = document.querySelectorAll('button');
+const display = document.getElementById('display');
+let result = '';
+let operand = '';
 
-- Modify this file only
-- The calculator should be completely functional
+const appendDigitToOperand = (button) => {
+    operand += button;
+    display.innerText = eval(operand);
+};
 
-*/
+const clearOperand = () => {
+    operand = '';
+};
+
+const clearResult = () => {
+    operand = '';
+    result = '';
+};
+
+const clearDisplay = () => {
+    display.innerText = eval('0');
+};
+
+const updateOperation = (button) => {
+    if (button === '+' || button === '-') {
+        if (!operand) {
+            result += `${operand}${button}`;
+        } else {
+            result += `${operand}${button}`;
+            display.innerText = (eval(result.slice(0, result.length - 1)));
+            clearOperand();
+        }
+    }
+    if (button === 'X') {
+        if (!operand) {
+            result += `0${operand}*`;
+            display.innerText = (eval(result.slice(0, result.length - 1)));
+        } else {
+            result += `${operand}*`;
+            display.innerText = (eval(result.slice(0, result.length - 1)));
+            clearOperand();
+        }
+    }
+    if (button === '/') {
+        if (!operand) {
+            result += `0${operand}/`;
+            display.innerText = (eval(result.slice(0, result.length - 1)));
+        } else {
+            result += `${operand}/`;
+            display.innerText = (eval(result.slice(0, result.length - 1)));
+            clearOperand();
+        }
+    }
+};
+
```

Comment:
Just a small suggestion for making this a bit DRYer 😬
```suggestion
function updateEvalAndClear(update, shouldClearOperand = true) {
    result += update;
    display.innerText = (eval(result.slice(0, result.length - 1)));
    shouldClearOperand && clearOperand();
}

const updateOperation = (button) => {
    if (button === '+' || button === '-') {
        const newResult = `${operand}${button}`
        if (operand) return updateEvalAndClear(newResult)
        result += newResult;
    }
    if (button === 'X') {
        if (operand) return updateEvalAndClear(`${operand}*`)
        updateEvalAndClear(`0${operand}*`, false)
    }
    if (button === '/') {
        if (operand) return updateEvalAndClear(`${operand}/`)
        updateEvalAndClear(`0${operand}/`, false);
    }
};

```

Reply:
Ooohh this looks way better, thanks for such illustrating code!